main/main.c

```diff
@@ -102,6 +140,72 @@ rte_strdup(const char *type, const char *s)
 	return strcpy(res, s);
 }
 
+FILE *log_file = NULL;
+struct rte_timer log_timer;
+
+static int
+cleanup_log(void)
+{
+	if (log_file != NULL) {
+		fclose(log_file);
+		log_file = NULL;
+	}
+
+	rte_free(log_base_dir);
+	rte_free(log_file_name_format);
+
+	return 0;
+}
+
+int
+gatekeeper_log_init(void)
+{
+	int ret;
+	time_t now;
+	struct tm *time_info;
+	char log_file_name[100];
+	char log_file_path[100];
+	FILE *new_log_file = NULL;
+
+	/*
+	 * Initialize log_file with stderr to guarantee that log_file
+	 * has a valid value even before a file is open.
+	 */
+	log_file = stderr;
```

Comment:
It's better to initialize `log_file` with `stderr`, so the comment is true even before `gatekeeper_log_init()` is called. Moreover, right now, when `gatekeeper_log_init()` is called for the second time or later, the previous log file is NOT being closed because the reference to it is lost at the assignment above.

Reply:
If we initialize ``log_file`` with ``stderr``, there will be an error: ``main/main.c:131:18: error: initializer element is not constant
 FILE *log_file = stderr;``